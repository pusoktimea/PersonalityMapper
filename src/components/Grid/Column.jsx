import React, {Component} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './column.scss';

class Column extends Component {
  static propTypes = {
    width: PropTypes.number.isRequired,
    children: PropTypes.node.isRequired,
    className: PropTypes.string
  };

  render() {
    const {width, children, className} = this.props;
    const baseClass = 'activate-column';
    return (
      <div
        className={cx(
          `${baseClass}`,
          `${baseClass}--${width}`,
          className
        )}
        {...this.props}
      >
        {children}
      </div>
    );
  }
}

export default Column;
