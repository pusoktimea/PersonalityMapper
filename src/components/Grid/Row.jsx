import React, {Component} from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import './row.scss';

class Row extends Component {
  static propTypes = {
    columnCount: PropTypes.number.isRequired,
    children: PropTypes.node.isRequired,
    className: PropTypes.string
  };

  render() {
    const {columnCount, children, className} = this.props;
    const baseClass = 'persmap-row';
    return (
      <div
        className={cx(
          baseClass,
          `${baseClass}--${columnCount}`,
          className
        )}
        {...this.props}
      >
        {children}
      </div>
    );
  }
}

export default Row;
