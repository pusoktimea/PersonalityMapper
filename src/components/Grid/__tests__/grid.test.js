import React from 'react';
import renderer from 'react-test-renderer';

import Row from '../Row';
import Column from '../Column';

describe('Grid test', () => {
  it('renders and matches snapshot', () => {
    expect(
      renderer.create(
        <Row columnCount={2}>
          <Column width={6}>test column</Column>
          <Column width={6}>test column</Column>
        </Row>
      ).toJSON()
    ).toMatchSnapshot();
  });
});
