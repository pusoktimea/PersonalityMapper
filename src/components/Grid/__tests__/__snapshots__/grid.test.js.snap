// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid test renders and matches snapshot 1`] = `
<div
  className="persmap-row persmap-row--2"
  columnCount={2}
>
  <div
    className="persmap-column persmap-column--6"
    width={6}
  >
    test column
  </div>
  <div
    className="persmap-column persmap-column--6"
    width={6}
  >
    test column
  </div>
</div>
`;
